pipeline-runner/R/qc-helpers.R

```diff
@@ -1,3 +1,42 @@
+#' Title
+#'
+#' @param scdata
+#'
+#' @return
+#' @export
+#'
+#' @examples
+generate_first_step_ids <- function(scdata) {
+  cells_id <- list()
+  for (sample_id in unique(scdata$samples)) {
+    cells_id[[sample_id]] <- scdata$cells_id[scdata$samples == sample_id]
+  }
+  return(cells_id)
+}
+
+#
+# Returns the positions to keep based on scdata and number of cells to keep
+#
+get_positions_to_keep <- function(scdata, num_cells_to_downsample) {
+  # Downsample plotData
+  num_cells_to_downsample <- downsample_plotdata(ncol(scdata), num_cells_to_downsample)
+  set.seed(123)
+  cells_position_to_keep <- sample(1:ncol(scdata), num_cells_to_downsample, replace = FALSE)
+  cells_position_to_keep <- sort(cells_position_to_keep)
+
+  return(cells_position_to_keep)
+}
+
+#
+# subset_ids subsets a seurat object with the cell ids
+#
+subset_ids <- function(scdata, cells_id) {
```

Comment:
wondering if we should differentiate the fact that `cells_id` is usually a list of vectors of `cells_id` for each sample but here it is just a vector of `cells_id`. Might want to rename either here or elsewhere.